src/engine/engineworkerscheduler.cpp

```diff
@@ -29,17 +32,17 @@ void EngineWorkerScheduler::addWorker(EngineWorker* pWorker) {
 
 void EngineWorkerScheduler::runWorkers() {
     // Wake the scheduler if we have written a worker-ready message to the
-    // scheduler. There is no race condition in accessing this boolean because
-    // both workerReady and runWorkers are called from the callback thread.
-    if (m_bWakeScheduler) {
-        m_bWakeScheduler = false;
+    // scheduler. This is called from the callback thread, so we use an
+    // atomic and not a mutex.
+    if (m_bWakeScheduler.exchange(false)) {
         m_waitCondition.wakeAll();
     }
 }
 
 void EngineWorkerScheduler::run() {
     static const QString tag("EngineWorkerScheduler");
-    while (!m_bQuit) {
+    bool quit = false;
+    while (!quit) {
         Event::start(tag);
         {
             const auto locker = lockMutex(&m_mutex);
```

Comment:
Super nitpick here, just to keep it consistent with the other scope's lock guard name
```suggestion
            const auto lock = lockMutex(&m_mutex);
```

Reply:
Oh I copy-pasted this from existing code. No preference myself.

Reply:
Ah, no, I copied this from Qt example code. Fixed!